### ISSUE TITLE
add steps for installing cachix on homepage

### ISSUE THREAD

**Issue by cdepillabout** [cachix/cachix#218]
I wanted to play around with [`static-haskell-nix`](https://github.com/nh2/static-haskell-nix), and the [instructions](https://github.com/nh2/static-haskell-nix#binary-caches-for-faster-building-optional) there recommend using cachix.

The instructions on `static-haskell-nix` don't explain how to install cachix, but just link to the cachix [homepage](https://cachix.org/).

The cachix homepage doesn't explain how to install cachix.  I tried clicking `Get Started ->`, but it is asking me to sign in to do OAuth with Github.  It is unclear whether I need to do this just to install `cachix` and use someone else's cache.

So I tried to install cachix on my machine, and run `cachix use static-haskell-nix` as recommended by the `static-haskell-nix` instructions:

```console
$ nix-env -iA nixos.cachix
...
$ cachix use static-haskell-nix

/etc/nixos/cachix: createDirectory: permission denied (Permission denied)
```

I can see that is is not able to create `/etc/nixos/cachix`.  At this point, I'm not sure why this is, but my guess is that it is trying to store some sort of configuration there.

I then thought to try nixpkgs `master` and install the latest version of `cachix`.  Maybe it would work.  However, it appears that the latest version of `cachix` (0.2.1) is marked broken in nixpkgs `master`.

So I thought maybe I need to run `cachix` with `sudo`.  This appears to work.

-------------------------------------------------------------------------------------------------------

I guess my solutions about how to fix this would be the following:

- Add a section to the `cachix` homepage explaining how to install `cachix`.  Ideally this would be directly on the homepage, and you wouldn't have to login first.
- Make the error from `cachix use` a little easier to understand.  It should suggest re-running with `sudo`, and a short explanation of what it is trying to do.

Ping @nh2, since this has to do with `static-haskell-nix`.

**Comment by cdepillabout** [comment]
Looks like the error message from `cachix use` is brought up on https://github.com/cachix/cachix/issues/186 as well.

**Comment by domenkozar** [comment]
The instructions should point to https://static-haskell-nix.cachix.org/

> I can see that is is not able to create /etc/nixos/cachix

Cachix writes down NixOS configuration, you can give is `-d PATH` to avoid `/etc/nixos` path.

> I then thought to try nixpkgs master and install the latest version of cachix

As per instructions, nixpkgs unstable is not officially supported as updates of ecosystem there are too rapid.

I agree about #186, I'll try to fix that one :)

**Comment by cdepillabout** [comment]
Thanks for the quick response.

> The instructions should point to https://static-haskell-nix.cachix.org/

Thanks! This is the type of instructions I was looking for.  I created https://github.com/nh2/static-haskell-nix/issues/49 to fix this in `static-haskell-nix`.

> nixpkgs unstable is not officially supported as updates of ecosystem there are too rapid

No worries, I'm completely fine installing it with something like `nix-env -iA cachix -f https://cachix.org/api/v1/install`.  It is just that I didn't see those instructions from the homepage.  Although I am fine with them being on a page like https://static-haskell-nix.cachix.org/.


I'd be happy if you wanted to close this, since my problems should mostly be solved with #186 and https://github.com/nh2/static-haskell-nix/issues/49.
